pkg/api/types.go

```diff
@@ -265,8 +265,8 @@ type GitRepoVolumeSource struct {
 // The contents of the target Secret's Data field will be presented in a volume
 // as files using the keys in the Data field as the file names.
 type SecretVolumeSource struct {
-	// Reference to a Secret
-	Target ObjectReference `json:"target"`
+	// Name of the secret in the pod's namespace to use
+	Target string `json:"target"`
```

Comment:
I am fine with the change, I just want to be sure we're not thrashing back and forth between modes.  If this is what you think is best, I'll go with it.